### PR TITLE
Removed auto ssh key gen if not supplied. Variable is now required.

### PR DESCRIPTION
**Pull request by thpang**

### CODE REVIEW

File: docs/CONFIG-VARS.md [hunk 1]
```diff
@@ -7,7 +7,6 @@ Supported configuration variables are listed in the table below.  All variables
 - [List of valid configuration variables](#list-of-valid-configuration-variables)
   - [Table of Contents](#table-of-contents)
   - [Required Variables](#required-variables)
-    - [Application](#application)
```
Comment (manoatsas) [inline docs/CONFIG-VARS.md:10]: Is there a reason as to why the link to "Application" was removed?

File: docs/CONFIG-VARS.md [hunk 2]
```diff
@@ -28,13 +27,11 @@ Terraform input variables can be set in the following ways:
 
 ## Required Variables
 
-### Application
```
Comment (manoatsas) [inline docs/CONFIG-VARS.md:31]: Ditto

File: docs/CONFIG-VARS.md [hunk 3]
```diff
@@ -28,13 +27,11 @@ Terraform input variables can be set in the following ways:
...
-
 | Name | Description | Type | Default | Notes |
 | :--- | ---: | ---: | ---: | ---: |
 | prefix | A prefix used in the name of all the Azure resources created by this script. | string | | The prefix string must start with a lowercase letter and contain only alphanumeric characters and dashes (-), but cannot end with a dash. |
 | location | The Azure Region to provision all resources in this script | string | "East US" | |
-| tags | Map of common tags to be placed on all Azure resources created by this script | map | { project_name = "sasviya4", environment = "dev" } | |
```
Comment (manoatsas) [inline docs/CONFIG-VARS.md:37]: Ditto again, "tags" I believe are required for some Azure resources

**Review by manoatsas — CHANGES_REQUESTED** [review]

File: docs/CONFIG-VARS.md (hunk 3 above)
Comment (manoatsas) [reply docs/CONFIG-VARS.md:37]: Looks like "tags" can be left out of required section.

File: docs/CONFIG-VARS.md (hunk 1 above)
Comment (thpang) [reply docs/CONFIG-VARS.md:10]: Yes, to keep it consistent with the AWS repo. If it's need we need to make changes across all providers.

File: docs/CONFIG-VARS.md (hunk 2 above)
Comment (thpang) [reply docs/CONFIG-VARS.md:31]: Same as above.

File: docs/CONFIG-VARS.md (hunk 3 above)
Comment (thpang) [reply docs/CONFIG-VARS.md:37]: So "tags" have are now in the "General" section.

**Pull request merged**